ME2Randomizer/Classes/Randomizers/ME2/Misc/RTexts.cs

```diff
@@ -167,11 +172,324 @@ private static void UwuifyTalkFile(TalkFile tf, bool keepCasing, List<int> skipI
                 }
 
                 var str = sb.ToString();
-                str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+
+                if (addReactions)
+                {
+                    str = AddReactionToLine(strData, str, keepCasing);
+                }
+                else //reactions not enabled, use original uwuify UwU insertion
+                { 
+                    str = str.Replace("fuck", keepCasing ? "UwU" : "uwu", StringComparison.InvariantCultureIgnoreCase);
+                }
+
                 TLKHandler.ReplaceString(sref.StringID, str, langCode);
             }
         }
 
+        private static string AddReactionToLine(string vanillaLine, string modifiedLine, bool keepCasing)
+        {
+            string finalString = "";
+            bool dangerousLine = false;
+
+            if (modifiedLine.Length < 2 || Regex.Matches(modifiedLine, "[a-zA-Z]").Count == 0 || vanillaLine.Contains('{'))
+            {
+                //I should go.
+                return modifiedLine;
+            }
+
+            string dangerousCharacters = "<\n";
+            if (modifiedLine.IndexOfAny(dangerousCharacters.ToCharArray()) >= 0 || vanillaLine.Length > 200)
+            {
+                dangerousLine = true;
+            }
+
+            //initialize reactions if they aren't loaded
+            if (ReactionList == null)
+            {
+                string rawReactionDefinitions = MERUtilities.GetEmbeddedStaticFilesTextFile("reactiondefinitions.xml");
+                StringReader reactionReader = new StringReader(rawReactionDefinitions);
+                XmlSerializer serializer = new XmlSerializer(typeof(List<Reaction>), new XmlRootAttribute("ReactionDefinitions"));
+                ReactionList = (List<Reaction>)serializer.Deserialize(reactionReader);
+            }
+
+            //split strings into sentences for processing
+            string pattern = @"(?<![M| |n][M|D|r][s|r]\.)(?<!(,""))(?<=[.!?""])(?= [A-Z])";
+            List<string> splitVanilla = new List<string>();
+            List<string> splitModified = new List<string>();
+
+            MatchCollection regexMatches = Regex.Matches(vanillaLine, pattern);
+            int modOffset = 0;
+
+            //for each regex match in the vanilla line:
+            for (int y = 0; y <= regexMatches.Count; y++)
```

Comment:
For code readability, I would suggest renaming y to something more useful. Like matchIndex or something.